customer-service/src/main/scala/domain/services/CustomerService.scala

```diff
@@ -4,17 +4,37 @@ import cats.MonadThrow
 import cats.data.OptionT
 import cats.syntax.all._
 import domain.repositories.CustomerRepository.CustomerRepository
-import domain.{Customer, CustomerId, CustomerWithDetails}
+import domain.{Customer, CustomerDetails, CustomerId, CustomerWithDetails}
 
 class CustomerService[F[_]: MonadThrow](repository: CustomerRepository[F])(
-    customerDetailsService: CustomerDetailsService[F]
+    customerDetailsServices: CustomerDetailsService[F]*
 ) {
   def create(customer: Customer): F[Unit]            = repository.add(customer)
   def createBatch(customers: Set[Customer]): F[Unit] = repository.addBatch(customers)
 
+  // not really sure of the pattern but the idea is to be be able to unplug the legacy customer details services by
+  // just not injecting it
+  // Also not very sure of how to deal with OptionT when you actually need to match for the value inside
   def get(id: CustomerId): OptionT[F, CustomerWithDetails] = for {
-    customer        <- repository.get(id)
-    customerDetails <- customerDetailsService.getCustomerDetails(id.value)
-  } yield CustomerWithDetails(customer.id, customer.name, customerDetails)
+    customer <- repository.get(id)
+    customerWithDetails <- OptionT.liftF(
+      getCustomerDetails(id.value, customerDetailsServices.toList.sortBy(_.getExecutionPriority)).value.map {
```

Comment:
read that comments first
- https://github.com/josjkw/CustomerService/pull/5/files#r1536059805
- https://github.com/josjkw/CustomerService/pull/5/files#r1536047675

suppose now that u moved that logic to `CustomerDetailsService`, and u have priority on repository level, so every CustomerDetailsRepository has its priority

in the issue itself: https://github.com/josjkw/CustomerService/issues/4 I mentioned that:
- CustomerLegacyDetails is very slow, it takes at least 10s to return data
- in CustomerLegacyDetails u have around 2% of all customerDetails, so 98% is already in new CustomerDetails service

suppose that we change conditions a little bit:
- traffic: legacy is 98% and new one is 2%
- each day legacy is smaller and new one is bigger, lets put is 1% every day
- legacy takes 10s, new one takes 5s

could u tell me what problems do u c?

what is average time to get details for customers in new faster service in design that u suggested?

can u come with any other design or u need a hint?